src/persist/src/storage.rs

```diff
@@ -16,8 +16,46 @@ use std::str::FromStr;
 
 use abomonation_derive::Abomonation;
 use async_trait::async_trait;
+use futures_executor::block_on;
 
 use crate::error::Error;
+use crate::indexed::encoding::BlobMeta;
+
+/// Sanity check whether we can decode the Blob's persisted meta object.
+///
+/// TODO: this is a hack and we will need to get rid of this once we have a
+/// proper backwards compatibility policy.
+pub fn check_meta_version<B: Blob>(b: &mut B) -> Result<(), Error> {
+    let meta = match block_on(b.get("META"))? {
+        None => return Ok(()),
+        Some(bytes) => bytes,
+    };
+
+    let current_version = BlobMeta::version();
+    let persisted_version = BlobMeta::encoded_version(&meta)
+        .ok_or_else(|| Error::from("unable to determine previously persisted encoding version."))?;
+
+    if current_version == persisted_version {
+        // Nothing to do here, everything is working as expected.
+        Ok(())
+    } else if current_version > persisted_version {
+        // Delete all the keys, as we are upgrading to a new version.
```

Comment:
we should probably log or do something else to indicate that we're deleting everything

Reply:
yeah for sure